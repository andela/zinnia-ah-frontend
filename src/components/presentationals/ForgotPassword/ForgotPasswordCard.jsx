import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';

import Title from '../Title/Title';
import Button from '../Button/Button';

import '../AuthenticationCard/AuthenticationCard.scss';
import '../Form.scss';

const ForgotPasswordCard = ({ email, onChange, submitRequest }) => {
  return (
    <div className="right">
      <div className="auth-card forgot-password">
        <Title content="Forgot Password?" className="text-center" />
        <hr className="hr" />
        <p>
          Enter the email you used to sign up and weâ€˜ll send you a password
          reset link
        </p>
        <Form onSubmit={submitRequest}>
          <Form.Field>
            <input
              placeholder="Email"
              name="email"
              type="email"
              value={email}
              onChange={onChange}
            />
          </Form.Field>
          <Button
            type="submit"
            value="SEND INSTRUCTIONS"
            className="btn-dark"
          />
        </Form>
      </div>
    </div>
  );
};

ForgotPasswordCard.propTypes = {
  email: PropTypes.string,
  onChange: PropTypes.func,
  submitRequest: PropTypes.func,
};

export default ForgotPasswordCard;
