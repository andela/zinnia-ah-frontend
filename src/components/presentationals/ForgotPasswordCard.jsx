import React from 'react';
import { Form } from 'semantic-ui-react';

import Title from './Title.jsx';
import Button from './Button.jsx';
import './AuthenticationCard.scss';
import './Form.scss';

const ForgotPasswordCard = () => {
  return (
    <div className="right">
      <div className="card forgot-password">
        <Title title={title} />
        <hr />
        <p>
          Enter the email you used to sign up and weâ€˜ll send you a password
          reset link
        </p>
        <Form>
          <Form.Field>
            <input placeholder="Email" type="email" />
          </Form.Field>
          <Button button={buttonValues} />
        </Form>
      </div>
    </div>
  );
};

const title = {
  content: 'Forgot Password?',
  class: 'text-center',
};

const buttonValues = {
  type: 'submit',
  value: 'SEND EMAIL',
  class: 'btn-dark',
};

export default ForgotPasswordCard;
