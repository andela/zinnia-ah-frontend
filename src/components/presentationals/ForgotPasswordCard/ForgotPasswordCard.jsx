import React from 'react';

import Title from '../Title/Title';
import Button from '../Button/Button.jsx';
import '../AuthenticationCard/AuthenticationCard.scss';
import '../Form.scss';

const ForgotPasswordCard = () => {
  return (
    <div className="right">
      <div className="card forgot-password">
        <Title title={title} />
        <hr className="hr" />
        <p>
          Enter the email you used to sign up and weâ€˜ll send you a password
          reset link
        </p>
        <form className="form">
          <div className="form-group">
            <input placeholder="Email" type="email" />
          </div>
          <Button button={buttonValues} />
        </form>
      </div>
    </div>
  );
};

const title = {
  content: 'Forgot Password?',
  class: 'text-center',
};

const buttonValues = {
  type: 'submit',
  value: 'SEND EMAIL',
  class: 'btn-dark',
};

export default ForgotPasswordCard;
