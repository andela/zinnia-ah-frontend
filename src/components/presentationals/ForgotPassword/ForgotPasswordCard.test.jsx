import React from 'react';
import { mount } from 'enzyme';

import ForgotPasswordCard from './ForgotPasswordCard';

describe('ForgotPasswordCard', () => {
  it('should render correctly', () => {
    const component = mount(<ForgotPasswordCard />);
    expect(component).toMatchSnapshot();
  });
});
