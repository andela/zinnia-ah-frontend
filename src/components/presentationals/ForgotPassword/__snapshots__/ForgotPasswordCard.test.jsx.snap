// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordCard should render correctly 1`] = `
<ForgotPasswordCard>
  <div
    className="right"
  >
    <div
      className="card forgot-password"
    >
      <Title
        title={
          Object {
            "class": "text-center",
            "content": "Forgot Password?",
          }
        }
      >
        <h1
          className="title undefined"
        />
      </Title>
      <hr
        className="hr"
      />
      <p>
        Enter the email you used to sign up and weâ€˜ll send you a password reset link
      </p>
      <form
        className="form"
      >
        <div
          className="form-group"
        >
          <input
            placeholder="Email"
            type="email"
          />
        </div>
        <Button
          button={
            Object {
              "class": "btn-dark",
              "type": "submit",
              "value": "SEND EMAIL",
            }
          }
        >
          <button
            className="btn undefined"
          />
        </Button>
      </form>
    </div>
  </div>
</ForgotPasswordCard>
`;
